src/legacy/server/saved_objects/service/lib/included_fields.ts

```diff
@@ -17,26 +17,30 @@
  * under the License.
  */
 
+function toArray(value: string | string[]): string[] {
+  return typeof value === 'string' ? [value] : value;
+}
 /**
  * Provides an array of paths for ES source filtering
- *
- * @param {string} type
- * @param {string|array} fields
- * @returns {array}
  */
-export function includedFields(type, fields) {
-  if (!fields || fields.length === 0) return;
+export function includedFields(type: string | string[] = '*', fields?: string[] | string) {
+  if (!fields || fields.length === 0) {
+    return;
+  }
 
   // convert to an array
-  const sourceFields = typeof fields === 'string' ? [fields] : fields;
-  const sourceType = type || '*';
+  const sourceFields = toArray(fields);
+  const sourceType = toArray(type);
 
-  return sourceFields
-    .map(f => `${sourceType}.${f}`)
+  return sourceType
+    .map(t => sourceFields.map(f => `${t}.${f}`))
+    .reduce((acc, value) => {
+      return acc.concat(value);
+    }, [])
     .concat('namespace')
     .concat('type')
     .concat('references')
     .concat('migrationVersion')
     .concat('updated_at')
-    .concat(fields); // v5 compatibility
+    .concat(fields); // v5 compatibility;
```

Comment:
Not sure why this line was changed to add a semicolon to the comment.

Reply:
🤷‍♂